csml_engine/src/db_connectors/dynamodb/conversations.rs

```diff
@@ -377,3 +377,167 @@ pub fn update_conversation(
 
     Ok(())
 }
+
+
+fn query_conversation(
+    client: &Client,
+    db: &mut DynamoDbClient,
+    limit: i64,
+    pagination_key: Option<HashMap<String, AttributeValue>>,
+) -> Result<QueryOutput, EngineError> {
+    let hash = Conversation::get_hash(client);
+
+    let expr_attr_names = [
+        ("#hashKey".to_string(), "hash".to_string()),
+        ("#rangeKey".to_string(), "range_time".to_string()),
+    ]
+    .iter()
+    .cloned()
+    .collect();
+
+    let expr_attr_values = [
+        (
+            String::from(":hashVal"),
+            AttributeValue {
+                s: Some(hash),
+                ..Default::default()
+            },
+        ),
+        (
+            String::from(":rangePrefix"),
+            AttributeValue {
+                s: Some(String::from("conversation#")),
+                ..Default::default()
+            },
+        ),
+    ]
+    .iter()
+    .cloned()
+    .collect();
+
+    let input = QueryInput {
+        table_name: get_table_name()?,
+        index_name: Some(String::from("TimeIndex")),
+        key_condition_expression: Some(
+            "#hashKey = :hashVal AND begins_with(#rangeKey, :rangePrefix)".to_owned(),
+        ),
+        expression_attribute_names: Some(expr_attr_names),
+        expression_attribute_values: Some(expr_attr_values),
+        limit: Some(limit),
+        exclusive_start_key: pagination_key,
+        scan_index_forward: Some(false),
+        select: Some(String::from("ALL_ATTRIBUTES")),
+        ..Default::default()
+    };
+
+    let future = db.client.query(input);
+    let data = db.runtime.block_on(future)?;
+
+    Ok(data)
+}
+
+pub fn delete_user_conversations(client: &Client, db: &mut DynamoDbClient) -> Result<(), EngineError> {
+    let mut pagination_key = None;
+
+    // retrieve all memories from dynamodb
+    loop {
+        let data = query_conversation(client, db, 25, pagination_key)?;
```

Comment:
no need to query the full items ("ALL_ATTRIBUTES"), we can use the specific attributes to get. See implementation here and replicate everywhere